File: examples/haskell/user_management.hs
{-# LANGUAGE OverloadedStrings #-}
import Database.PostgreSQL.Simple
main :: IO ()
main = do
    conn <- connect defaultConnectInfo { connectHost="127.0.0.1", connectPort=5408, connectUser="admin", connectPassword="admin123", connectDatabase="huntiandb" }
    xs <- query_ conn "SHOW USERS" :: IO [(String, String)]
    putStrLn "── Current Users ──"
    mapM_ (\(u,r) -> putStrLn $ "  " ++ u ++ " (" ++ r ++ ")") xs
    _ <- execute_ conn "INSERT INTO users (username, role) VALUES ('hs_analyst', 'reader')"
    putStrLn "[OK] INSERT INTO users: hs_analyst (reader)"
    close conn; putStrLn "[DONE]"

File: examples/rust/user_management.rs
// rust/user_management.rs — 用户管理
use std::process::Command;
fn exec(sql: &str) -> String {
    let out = Command::new("psql").args(["-h","127.0.0.1","-p","5408","-U","admin","-d","huntiandb","-c",sql]).env("PGPASSWORD","admin123").output().unwrap();
    String::from_utf8_lossy(&out.stdout).to_string()
}
fn main() {
    println!("{}", exec("SHOW USERS"));
    exec("INSERT INTO users (username, role) VALUES ('rust_analyst', 'reader')");
    println!("[OK] INSERT INTO users: rust_analyst (reader)");
    println!("[DONE]");
}
